File: 1 - metodosBusca/pacoteBusca_JomiHubner/src/grafo/ArestaValorada.java
package grafo;

public abstract class ArestaValorada extends Aresta {
    int custo;
		
    public ArestaValorada(Vertice vi, Vertice vj, int custo) {
        super(vi,vj);
        this.custo = custo;
    }

    public int getCusto() {
        return custo;
    }
    
    public String toString() {
		return vi + " <- "+custo+" -> " + vj;
    }
}

File: 1 - metodosBusca/python/app/core/grafo/aresta_valorada.py
from .aresta import Aresta

class ArestaValorada(Aresta):
    def __init__(self, vi, vj, custo):
        super().__init__(vi, vj)
        self.custo = custo

    def get_custo(self):
        return self.custo

    def __str__(self):
        return f"{self.vi} <- {self.custo} -> {self.vj}"